shm_wrapper/README.md

```diff
@@ -20,12 +13,59 @@ make test2
 ```
 Then run by opening two separate terminal windows and running first `./test1` in one window and then `./test2` in the other. The tests will run automatically to completion and print many status messages and updates to standard output. 
 
-If using a Linux system or Linux emulator on a Windows system, you need to set `LIBFLAGS` in the Makefile to `-lrt -lpthread`. For a Mac, set `LIBFLAGS=` to the empty string.
+`connect_static_devices.c` is a simple program that is, in some sense, a fake `dev_handler`, in that it creates the shared memory blocks and keeps the blocks open at some static value. This program is necessary if we want to test any other process that uses the shared memory wrapper by itself, without having to start up the full device handler process. Without it, the process we want to test would generate a segmentation fault (due to the shared memory not existing). To compile this program, run
```

Comment:
Forgot to ask you this but can we have it so it doesn't segfault and instead exits if SHM is not initialized? I did this for logger already.

Reply:
This is a good idea, yes. I'll change this behavior

Reply:
Wait I already put it in, the `error` function in the shared memory function already exits by default. I wasn't really sure why it was segfaulting even on exit.